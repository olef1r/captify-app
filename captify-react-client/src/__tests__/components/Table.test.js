import React from 'react';
import { shallow, mount } from 'enzyme';
import SimpleTable from '../../components/Table';

let wrapper;
const data =  ['1', '2'];

beforeEach(() => {
  wrapper = mount(<SimpleTable rows={data}/>);
});

describe('<Table - Component', () => {
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveDisplayName('SimpleTable');
    expect(wrapper).toContainMatchingElement('tr');
    expect(wrapper).toContainMatchingElement('table');
    expect(wrapper).toContainMatchingElement('tbody');
    expect(wrapper).toContainMatchingElement('th');
  });
});
