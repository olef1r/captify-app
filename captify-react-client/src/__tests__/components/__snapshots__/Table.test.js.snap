// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table - Component renders without crashing 1`] = `
<SimpleTable
  rows={
    Array [
      "1",
      "2",
    ]
  }
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
    maxWidth="xs"
  >
    <ForwardRef(Container)
      className="makeStyles-container-2"
      classes={
        Object {
          "fixed": "MuiContainer-fixed",
          "maxWidthLg": "MuiContainer-maxWidthLg",
          "maxWidthMd": "MuiContainer-maxWidthMd",
          "maxWidthSm": "MuiContainer-maxWidthSm",
          "maxWidthXl": "MuiContainer-maxWidthXl",
          "maxWidthXs": "MuiContainer-maxWidthXs",
          "root": "MuiContainer-root",
        }
      }
      maxWidth="xs"
    >
      <div
        className="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthXs"
      >
        <WithStyles(ForwardRef(Paper))
          className="makeStyles-root-3"
        >
          <ForwardRef(Paper)
            className="makeStyles-root-3"
            classes={
              Object {
                "elevation0": "MuiPaper-elevation0",
                "elevation1": "MuiPaper-elevation1",
                "elevation10": "MuiPaper-elevation10",
                "elevation11": "MuiPaper-elevation11",
                "elevation12": "MuiPaper-elevation12",
                "elevation13": "MuiPaper-elevation13",
                "elevation14": "MuiPaper-elevation14",
                "elevation15": "MuiPaper-elevation15",
                "elevation16": "MuiPaper-elevation16",
                "elevation17": "MuiPaper-elevation17",
                "elevation18": "MuiPaper-elevation18",
                "elevation19": "MuiPaper-elevation19",
                "elevation2": "MuiPaper-elevation2",
                "elevation20": "MuiPaper-elevation20",
                "elevation21": "MuiPaper-elevation21",
                "elevation22": "MuiPaper-elevation22",
                "elevation23": "MuiPaper-elevation23",
                "elevation24": "MuiPaper-elevation24",
                "elevation3": "MuiPaper-elevation3",
                "elevation4": "MuiPaper-elevation4",
                "elevation5": "MuiPaper-elevation5",
                "elevation6": "MuiPaper-elevation6",
                "elevation7": "MuiPaper-elevation7",
                "elevation8": "MuiPaper-elevation8",
                "elevation9": "MuiPaper-elevation9",
                "root": "MuiPaper-root",
                "rounded": "MuiPaper-rounded",
              }
            }
          >
            <div
              className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-3 MuiPaper-rounded"
            >
              <WithStyles(ForwardRef(Table))
                aria-label="simple table"
              >
                <ForwardRef(Table)
                  aria-label="simple table"
                  classes={
                    Object {
                      "root": "MuiTable-root",
                      "stickyHeader": "MuiTable-stickyHeader",
                    }
                  }
                >
                  <table
                    aria-label="simple table"
                    className="MuiTable-root"
                  >
                    <WithStyles(ForwardRef(TableHead))>
                      <ForwardRef(TableHead)
                        classes={
                          Object {
                            "root": "MuiTableHead-root",
                          }
                        }
                      >
                        <thead
                          className="MuiTableHead-root"
                        >
                          <WithStyles(ForwardRef(TableRow))>
                            <ForwardRef(TableRow)
                              classes={
                                Object {
                                  "footer": "MuiTableRow-footer",
                                  "head": "MuiTableRow-head",
                                  "hover": "MuiTableRow-hover",
                                  "root": "MuiTableRow-root",
                                  "selected": "Mui-selected",
                                }
                              }
                            >
                              <tr
                                className="MuiTableRow-root MuiTableRow-head"
                              >
                                <WithStyles(ForwardRef(TableCell))
                                  align="center"
                                >
                                  <ForwardRef(TableCell)
                                    align="center"
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTableCell-alignCenter",
                                        "alignJustify": "MuiTableCell-alignJustify",
                                        "alignLeft": "MuiTableCell-alignLeft",
                                        "alignRight": "MuiTableCell-alignRight",
                                        "body": "MuiTableCell-body",
                                        "footer": "MuiTableCell-footer",
                                        "head": "MuiTableCell-head",
                                        "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                                        "paddingNone": "MuiTableCell-paddingNone",
                                        "root": "MuiTableCell-root",
                                        "sizeSmall": "MuiTableCell-sizeSmall",
                                        "stickyHeader": "MuiTableCell-stickyHeader",
                                      }
                                    }
                                  >
                                    <th
                                      aria-sort={null}
                                      className="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-head"
                                      scope="col"
                                    >
                                      Items
                                    </th>
                                  </ForwardRef(TableCell)>
                                </WithStyles(ForwardRef(TableCell))>
                              </tr>
                            </ForwardRef(TableRow)>
                          </WithStyles(ForwardRef(TableRow))>
                        </thead>
                      </ForwardRef(TableHead)>
                    </WithStyles(ForwardRef(TableHead))>
                    <WithStyles(ForwardRef(TableBody))>
                      <ForwardRef(TableBody)
                        classes={
                          Object {
                            "root": "MuiTableBody-root",
                          }
                        }
                      >
                        <tbody
                          className="MuiTableBody-root"
                        >
                          <WithStyles(ForwardRef(TableRow))
                            key="0"
                          >
                            <ForwardRef(TableRow)
                              classes={
                                Object {
                                  "footer": "MuiTableRow-footer",
                                  "head": "MuiTableRow-head",
                                  "hover": "MuiTableRow-hover",
                                  "root": "MuiTableRow-root",
                                  "selected": "Mui-selected",
                                }
                              }
                            >
                              <tr
                                className="MuiTableRow-root"
                              >
                                <WithStyles(ForwardRef(TableCell))
                                  align="center"
                                  component="th"
                                  scope="row"
                                  variant="body"
                                >
                                  <ForwardRef(TableCell)
                                    align="center"
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTableCell-alignCenter",
                                        "alignJustify": "MuiTableCell-alignJustify",
                                        "alignLeft": "MuiTableCell-alignLeft",
                                        "alignRight": "MuiTableCell-alignRight",
                                        "body": "MuiTableCell-body",
                                        "footer": "MuiTableCell-footer",
                                        "head": "MuiTableCell-head",
                                        "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                                        "paddingNone": "MuiTableCell-paddingNone",
                                        "root": "MuiTableCell-root",
                                        "sizeSmall": "MuiTableCell-sizeSmall",
                                        "stickyHeader": "MuiTableCell-stickyHeader",
                                      }
                                    }
                                    component="th"
                                    scope="row"
                                    variant="body"
                                  >
                                    <th
                                      aria-sort={null}
                                      className="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-body"
                                      scope="row"
                                    >
                                      1
                                    </th>
                                  </ForwardRef(TableCell)>
                                </WithStyles(ForwardRef(TableCell))>
                              </tr>
                            </ForwardRef(TableRow)>
                          </WithStyles(ForwardRef(TableRow))>
                          <WithStyles(ForwardRef(TableRow))
                            key="1"
                          >
                            <ForwardRef(TableRow)
                              classes={
                                Object {
                                  "footer": "MuiTableRow-footer",
                                  "head": "MuiTableRow-head",
                                  "hover": "MuiTableRow-hover",
                                  "root": "MuiTableRow-root",
                                  "selected": "Mui-selected",
                                }
                              }
                            >
                              <tr
                                className="MuiTableRow-root"
                              >
                                <WithStyles(ForwardRef(TableCell))
                                  align="center"
                                  component="th"
                                  scope="row"
                                  variant="body"
                                >
                                  <ForwardRef(TableCell)
                                    align="center"
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTableCell-alignCenter",
                                        "alignJustify": "MuiTableCell-alignJustify",
                                        "alignLeft": "MuiTableCell-alignLeft",
                                        "alignRight": "MuiTableCell-alignRight",
                                        "body": "MuiTableCell-body",
                                        "footer": "MuiTableCell-footer",
                                        "head": "MuiTableCell-head",
                                        "paddingCheckbox": "MuiTableCell-paddingCheckbox",
                                        "paddingNone": "MuiTableCell-paddingNone",
                                        "root": "MuiTableCell-root",
                                        "sizeSmall": "MuiTableCell-sizeSmall",
                                        "stickyHeader": "MuiTableCell-stickyHeader",
                                      }
                                    }
                                    component="th"
                                    scope="row"
                                    variant="body"
                                  >
                                    <th
                                      aria-sort={null}
                                      className="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-body"
                                      scope="row"
                                    >
                                      2
                                    </th>
                                  </ForwardRef(TableCell)>
                                </WithStyles(ForwardRef(TableCell))>
                              </tr>
                            </ForwardRef(TableRow)>
                          </WithStyles(ForwardRef(TableRow))>
                        </tbody>
                      </ForwardRef(TableBody)>
                    </WithStyles(ForwardRef(TableBody))>
                  </table>
                </ForwardRef(Table)>
              </WithStyles(ForwardRef(Table))>
            </div>
          </ForwardRef(Paper)>
        </WithStyles(ForwardRef(Paper))>
      </div>
    </ForwardRef(Container)>
  </WithStyles(ForwardRef(Container))>
</SimpleTable>
`;
